docs/src/App.js

```diff
@@ -0,0 +1,61 @@
+import './index.css';
+import {useState} from 'react';
+import Header from './components/Header';
+import AddTask from './components/AddTask';
+import Tasks from './components/Tasks';
+import Footer from './components/Footer';
+
+
+
+
+function App(props) {
+    const [tasks, setTasks] = useState(props.initialData)
+    const [showComplete, setShowComplete] = useState(false)
+
+    // Delete task
+    function deleteTask () {
+        setTasks(tasks.filter((task) => task.complete !== true))
+    }
+
+    // Add task
+    function addTask (e) {
+        const complete = false
+        const hidden = false
+        const id = Math.floor(Math.random() * 10000) + 1
+        const newTask = {id,...e,complete}
+        setTasks([...tasks, newTask])
+    }
+
+    // Hide Task
+    function hideTask () {
+        setTasks(tasks.map((task) => task.complete
+            ? {...task, hidden: !task.hidden} : task))
```

Comment:
Unclear to me how hidden is used.  Why not just filter out the completed tasks:
tasksToShow = showComplete ? tasks.filter(t=>!t.completed) : tasks;